proposals/dao/v2Phase1-part1.ts

```diff
@@ -0,0 +1,535 @@
+import {
+  BondingCurve,
+  ERC20CompoundPCVDeposit,
+  EthBondingCurve,
+  RatioPCVController,
+  Timelock,
+  UniswapPCVDeposit
+} from '@custom-types/contracts';
+import { RunUpgradeFunc, SetupUpgradeFunc, TeardownUpgradeFunc, ValidateUpgradeFunc } from '@custom-types/types';
+import '@nomiclabs/hardhat-ethers';
+import { getImpersonatedSigner } from '@test/helpers';
+import chai from 'chai';
+import CBN from 'chai-bn';
+import hre, { ethers } from 'hardhat';
+
+before(() => {
+  chai.use(CBN(ethers.BigNumber));
+});
+
+// Genesis functionality (among other stuff) was removed in commit 195415, but obviously not removed from the on-chain core
+// Thus we need the original ABI
+const OriginalCoreABI = [
```

Comment:
This should be imported as an interface in a solidity file so that a compiler can generate that ABI. Sticking that in the codebase should be avoided.

Reply:
I don't feel strongly about this but its a good idea